crypto/context.c

```diff
@@ -374,6 +374,9 @@ static int default_context_inited = 0;
 
 DEFINE_RUN_ONCE_STATIC(default_context_do_init)
 {
+    if (!ossl_init_thread())
```

Comment:
It looks like `ossl_init_thread()` does an `CRYPTO_THREAD_init_local()`, so shouldn't there be a corresponding cleanup call (`ossl_cleanup_thread()`?) in the error return if either of the two subsequent initialisations (line 380 and 383) fail?